core/src/main/java/org/elasticsearch/index/mapper/SeqNoFieldMapper.java

```diff
@@ -252,11 +253,16 @@ public void postParse(ParseContext context) throws IOException {
         // we share the parent docs fields to ensure good compression
         SequenceIDFields seqID = context.seqID();
         assert seqID != null;
-        for (int i = 1; i < context.docs().size(); i++) {
+        int numDocs = context.docs().size();
+        final Version versionCreated = context.mapperService().getIndexSettings().getIndexVersionCreated();
```

Comment:
not related to this change, but I think we should have add `QueryShardContext#getIndexVersionCreated()` helper method that does this: `mapperService().getIndexSettings().getIndexVersionCreated()`.

Reply:
Arg... I meant `ParseContext#getIndexVersionCreated()`

Reply:
I will open a followup but I think we need to have a more common class across all index level contexts maybe a base class?